app/plugin_key_value_store.go

```diff
@@ -47,6 +48,35 @@ func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte,
 	return nil
 }
 
+func (a *App) CompareAndSetPluginKey(pluginId string, key string, oldValue, newValue []byte) (bool, *model.AppError) {
+	kv := &model.PluginKeyValue{
+		PluginId: pluginId,
+		Key:      key,
+		Value:    newValue,
+	}
+
+	resultChan := make(chan store.StoreResult, 1)
+	go func() {
+		updated, err := a.Srv.Store.Plugin().CompareAndUpdate(kv, oldValue)
+		resultChan <- store.StoreResult{Data: updated, Err: err}
+		close(resultChan)
+	}()
+
+	result := <-resultChan
+	updated := result.Data.(bool)
+	if result.Err != nil {
```

Comment:
Thanks for adopting the new pattern here. :+1: 

But I think you don't need the go routine here. You will wait for the result any way in L65, hence you can just call `updated, err := a.Srv.Store.Plugin().CompareAndUpdate(kv, oldValue)`.

```suggestion
	updated, err := a.Srv.Store.Plugin().CompareAndUpdate(kv, oldValue)
	if err != nil {
```

Reply:
@hanzei did you mean the go-routine is unnecessary on [L59](https://github.com/mattermost/mattermost-server/pull/10634/files#diff-afd8e004133130ca459a180c0ed896fdR59)?

if i call `updated, err := a.Srv.Store.Plugin().CompareAndUpdate(kv, oldValue)` here won't i be updating twice once here and once in [L60](https://github.com/mattermost/mattermost-server/pull/10634/files#diff-afd8e004133130ca459a180c0ed896fdR60)

Reply:
clarified offline, @hanzei did mean to not going through the channel as it's unnecessary.